composable_kernel/include/tensor_operation/gridwise_batched_gemm_xdlops_v2r3.hpp

```diff
@@ -0,0 +1,725 @@
+                                 m_thread_data_on_grid_idx[I1],
+                                 n_thread_data_on_grid_idx[I1],
+                                 m_thread_data_on_grid_idx[I2],
+                                 m_thread_data_on_grid_idx[I3],
+                                 m_thread_data_on_grid_idx[I4],
+                                 n_thread_data_on_grid_idx[I2]),
+                c_element_op};
+
+        auto a_grid_buf = make_dynamic_buffer<AddressSpaceEnum_t::Global>(
+            p_a_grid_, a_grid_desc_k0_m_k1.GetElementSpaceSize());
+
+        auto c_grid_buf = make_dynamic_buffer<AddressSpaceEnum_t::Global>(
+            p_c_grid_, c_grid_desc_m0_n0_m1_n1_m2_m3_m4_n2.GetElementSpaceSize());
+
+        int bb = 0;
+        do
+        {
+            // preload data into LDS
+            {
+                a_blockwise_copy.RunRead(a_grid_desc_k0_m_k1, a_grid_buf);
+                b_blockwise_copy.RunRead(b_grid_desc_k0_n_k1, b_grid_buf);
+
+                a_blockwise_copy.RunWrite(a_block_desc_k0_m_k1, a_block_buf);
+                b_blockwise_copy.RunWrite(b_block_desc_k0_n_k1, b_block_buf);
+            }
+
+            // initialize c_thread_buf
+            c_thread_buf.Clear();
+
+            if constexpr(HasMainKBlockLoop)
+            {
+                // main body
+                index_t k_block_data_begin = 0;
+                do
+                {
+                    a_blockwise_copy.MoveSrcSliceWindow(a_grid_desc_k0_m_k1,
+                                                        a_block_slice_copy_step);
+                    b_blockwise_copy.MoveSrcSliceWindow(b_grid_desc_k0_n_k1,
+                                                        b_block_slice_copy_step);
+
+                    a_blockwise_copy.RunRead(a_grid_desc_k0_m_k1, a_grid_buf);
+
+                    block_sync_lds();
+
+                    b_blockwise_copy.RunRead(b_grid_desc_k0_n_k1, b_grid_buf);
+
+                    blockwise_gemm.Run(a_block_buf, b_block_buf, c_thread_buf);
+
+                    block_sync_lds();
+
+                    a_blockwise_copy.RunWrite(a_block_desc_k0_m_k1, a_block_buf);
+                    b_blockwise_copy.RunWrite(b_block_desc_k0_n_k1, b_block_buf);
+
+                    k_block_data_begin += K0PerBlock;
+                } while(k_block_data_begin < (K0 - K0PerBlock));
+            }
+
+            // tail
+            {
+                block_sync_lds();
+
+                blockwise_gemm.Run(a_block_buf, b_block_buf, c_thread_buf);
+            }
+
+            // output: register to global memory
+            {
+
+                c_thread_copy.Run(c_m0_n0_m1_n1_m2_m3_m4_n2_thread_desc,
+                                  make_tuple(I0, I0, I0, I0, I0, I0, I0, I0),
+                                  c_thread_buf,
+                                  c_grid_desc_m0_n0_m1_n1_m2_m3_m4_n2,
+                                  c_grid_buf);
+            }
+
+            a_grid_buf.p_data_ += a_batch_stride;
```

Comment:
So p_data_ is updated here

Is buffer resource in a_grid_buf_ also updated? If not, this is a bug.

If you are not sure what is "buffer resource", please talk with me

Reply:
a_grid_buf itself is not updated. I used to construct a new buffer inside the outer loop, and then you suggested moving it outside and updating it member p_data_, which reduces branches.

Reply:
Added element_space_size_ update